azurerm/internal/services/compute/tests/linux_virtual_machine_resource_scaling_test.go

```diff
@@ -66,6 +66,31 @@ func TestAccLinuxVirtualMachine_scalingDedicatedHost(t *testing.T) {
 	})
 }
 
+func TestAccLinuxVirtualMachine_scalingDedicatedHostUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_linux_virtual_machine", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: checkLinuxVirtualMachineIsDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: testLinuxVirtualMachine_scalingDedicatedHost(data),
+				Check: resource.ComposeTestCheckFunc(
+					checkLinuxVirtualMachineExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testLinuxVirtualMachine_scalingDedicatedHostUpdate(data),
+				Check: resource.ComposeTestCheckFunc(
+					checkLinuxVirtualMachineExists(data.ResourceName),
+				),
+			},
```

Comment:
since we have support for removing the Dedicated Host ID, can we add a test for starting without a dedicated host, then using one and then not using a dedicated host?

we're also missing an import test at the end here?

Reply:
Test added and fixed